mm/mm_heap/mm_initialize.c

```diff
@@ -93,12 +93,17 @@ void mm_addregion(FAR struct mm_heap_s *heap, FAR void *heapstart,
 
   DEBUGVERIFY(mm_takesemaphore(heap));
 
-  /* Adjust the provided heap start and size so that they are both aligned
-   * with the MM_MIN_CHUNK size.
+  /* Adjust the provided heap start and size.
+   *
+   * Note: (uintptr_t)node + SIZEOF_MM_ALLOCNODE is what's actually
+   * returned to the malloc user, which should have natural alignment.
+   * (that is, in this implementation, MM_MIN_CHUNK-alignment.)
    */
 
-  heapbase = MM_ALIGN_UP((uintptr_t)heapstart);
-  heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize);
+  heapbase = MM_ALIGN_UP((uintptr_t)heapstart + SIZEOF_MM_ALLOCNODE)
+             - SIZEOF_MM_ALLOCNODE;
+  heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize)
```

Comment:
```suggestion
  heapend  = MM_ALIGN_DOWN((uintptr_t)heapstart + (uintptr_t)heapsize) - SIZEOF_MM_ALLOCNODE;
```
This is also a solution to make heapsize align with MM_MIN_CHUNK and it seems better.

Reply:
i don't understand your suggestion. the original patch already has `- SIZEOF_MM_ALLOCNODE`.

Reply:
@yamt Sorry, i missed some code. So we satisfy the goal of #5991 and #5906 at the same time. Look good to me.